test/Algebra/Graph/Test/Undirected.hs

```diff
@@ -64,9 +66,25 @@ testUndirected = do
     test "edgeCount   . fromUndirected <= (*2) . edgeCount" $ \(x :: G) ->
           (G.edgeCount . fromUndirected) x <= ((*2) . edgeCount) x
 
+    putStrLn $ "\n============ Graph.Undirected.complement ================"
+
+    test "complement empty              == empty" $
+          complement empty              == (empty :: UGI)
+
+    test "complement (vertex 1)         == (vertex 1)" $
```

Comment:
The properties are stated for arbitrary values `x`, `y` and `z` but the tests are only for fixed values for some reason. Let's QuickCheck the properties exactly as they are written in Haddock.